nvtabular/ops/data_stats.py

```diff
@@ -13,95 +13,290 @@
+                        num_zeros=dask_stats[col]["num_zeroes"].item(),
+                    )
+                )
+                feature.type = 1 if np.issubdtype(dtype, np.floating) else 0
+            elif dtype == "object":
+                feature.string_stats.CopyFrom(
+                    statistics_pb2.StringStatistics(
+                        common_stats=common_stats,
+                        avg_length=dask_stats[col]["avg_str_length"],
+                        unique=dask_stats[col]["nunique"].item(),
+                    )
+                )
+                feature.type = 2
+
+                ranks = feature.string_stats.rank_histogram
+                for ind, (val, freq) in enumerate(
+                    dask_stats[col]["top_values"].to_pandas().items()
+                ):
+                    f = feature.string_stats.top_values.add()
+                    f.value = val
+                    f.frequency = freq
+                    b = ranks.buckets.add()
+                    b.CopyFrom(
+                        statistics_pb2.RankHistogram.Bucket(
+                            low_rank=ind, high_rank=ind, label=val, sample_count=freq
+                        )
+                    )
+
+        if self.cross_columns:
+            corr, cov = dask_stats["corr"], dask_stats["cov"]
+            for (path_x, cov_val), (_, corr_val) in zip(cov.items(), corr.items()):
+                for (path_y, covariance), correlation in zip(cov_val.items(), corr_val.items()):
+                    cross = statistics_pb2.CrossFeatureStatistics(
+                        path_x=path_pb2.Path(step=[path_x]),
+                        path_y=path_pb2.Path(step=[path_y]),
+                        num_cross_stats=statistics_pb2.NumericCrossStatistics(
+                            correlation=float(correlation), covariance=float(covariance)
+                        ),
+                    )
+                    c = self.stats.cross_features.add()
+                    c.CopyFrom(cross)
+
+        if self.stats_path:
+            self.stats_list().save(self.output_path)
 
     def clear(self):
-        self.output = {}
+        self.schema = None
+
+    def stats_list(self, others=None) -> "DatasetCollectionStatistics":
+        if not others:
+            others = []
+        from tensorflow_metadata.proto.v0 import statistics_pb2
+
+        data = statistics_pb2.DatasetFeatureStatisticsList()
+
+        d = data.datasets.add()
+        d.CopyFrom(self.stats)
+
+        for d in others:
+            x = data.datasets.add()
+            x.CopyFrom(d)
+
+        return DatasetCollectionStatistics(data)
+
+    def to_schema(self, tags_by_column):
+        self.tags_by_column = tags_by_column
+
+        return self.prepare_schema(self.stats)
 
     transform.__doc__ = Operator.transform.__doc__
     fit.__doc__ = StatOperator.fit.__doc__
     fit_finalize.__doc__ = StatOperator.fit_finalize.__doc__
+
+
+class DatasetCollectionStatistics:
```

Comment:
Since this for rendering, I wonder if we should split it out into a separate sub-package (e.g. `notebooks` or `rendering_utils`)